### PR TITLE
🦧 Form Slider

### LINKED ISSUE #57: Form slider
## Description

Style and wrap @reach/slider to make a slider component:

image

Note that this doesn't including styling the labels, just the sliders themselves.

All of the styling can be done via styled components and selectors.

Branch name: `form-slider`

## Todo

- Gain a thorough understanding of Reach's Slider component (read through the entire page)
- Create a new Slider component in `ui/Slider.js`.
- Get Reach's slider appearing on the page
- Style the slider track itself
- Make each step appear as a discrete circle
  - SliderMarker
- Style the currently selected step
  - Use data attributes to do this
- Add support for labels underneath each step
  - This will have to be a custom prop passed into your Slider (that doesn't get passed into Reach's Slider), an array of labels
  - Each label gets rendered underneath each step
  - Perhaps you can achieve this by rendering the labels equally spaced under the slider
  - But it's probably better to wrap SlideerMarker and render some text under it

## Useful links

- https://reach.tech/slider/
- Figma
- Styled Components docs
- https://reach.tech/styling/

### PR DESCRIPTION
**Pull request by Archna-1**
## Description
Create a slider that takes in and displays labels and passes back information based on the location of the slider handle.

Fixes #57 

## Todos

- [x] Store information based on the slider handle's location
- [x] Style slider markers into circles
- [x] Center labels

## Screenshots
Screen Shot 2021-04-03 at 1 02 50 PM

### CODE REVIEW

**Comment by Qrtn** [comment]
Wow Archna, so fast!! Nice work! Will review before dev night today. Can you make the slider ticks circles?

**Comment by Archna-1** [comment]
> Wow Archna, so fast!! Nice work! Will review before dev night today. Can you make the slider ticks circles?

Yes! I actually did that, but I didn't change the photo yet haha

**Comment by Qrtn** [comment]
Awesome! And it looks like the labels still aren't centered under each tick -- any thoughts on how to fix that?

**Review by Qrtn — APPROVED** [review]
Looks good to me Archna, great work!!

File: src/components/mobile/SettingsPage.js [hunk 1]
```diff
@@ -63,6 +64,12 @@ const SettingsPage = () => (
         </AccordionPanel>
       </AccordionItem>
     </SettingsAccordion>
+
+    <Slider
+      style={{ marginLeft: 40, marginRight: 40 }}
```
Comment (SirajChokshi) [inline src/components/mobile/SettingsPage.js:69]: Consider making this one rule `{{ margin: '0 40px' }}`

File: src/components/ui/Slider.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+`
+
+// We intentionally omit SliderRange because we don't want to highlight parts of the track
+const Slider = ({ labels, ...props }) => (
+  <StyledSliderInput max={labels.length - 1} {...props}>
+    <SliderTrack>
+      {labels.map((label, index) => (
+        <SliderMarker key={index} value={index}>
+          <div>{label}</div>
```
Comment (SirajChokshi) [inline src/components/ui/Slider.js:54]: It might be more semantic to use a label tag here instead of the div

**Review by SirajChokshi — APPROVED** [review]
Looks good! Great job commenting the styles.

File: src/components/ui/Slider.js (hunk 2 above)
Comment (Qrtn) [reply src/components/ui/Slider.js:54]: Let's stick to div here. I think labels need to be associated with an input. https://stackoverflow.com/questions/33214093/is-it-good-practice-to-use-label-for-non-input-non-interactive-elements

File: src/components/mobile/SettingsPage.js (hunk 1 above)
Comment (Archna-1) [reply src/components/mobile/SettingsPage.js:69]: So, this is still in-line style but style={{ margin: '0 40px' }} is what you're saying, right?

Comment (Archna-1) [reply src/components/mobile/SettingsPage.js:69]: Okay I didn't look at the file title lmao my bad. You mean to the slider itself lol

**Pull request merged**